tests/rhui3_tests/test_CLI.py

```diff
@@ -0,0 +1,112 @@
+
+with open('/tmp/rhui3-tests/tests/rhui3_tests/tested_repos.yaml', 'r') as file:
+    doc = yaml.load(file)
+
+yum_repo_name_1 = doc['CLI_repo1']['name']
+yum_repo_id_1 = doc['CLI_repo1']['id']
+yum_repo_name_2 = doc['CLI_repo2']['name']
+yum_repo_id_2 = doc['CLI_repo2']['id']
+
+custom_repo_name = "my_custom_repo"
+
+def setUp():
+    print "*** Running %s: *** " % basename(__file__)
+
+def test_01_initial_run():
+    '''Do an initial rhui-manager run to make sure we are logged in'''
+    RHUIManager.initial_run(connection)
+
+def test_02_check_empty_repo_list():
+    '''Check if the repolist is empty (interactively; not currently supported by the CLI)'''
+    nose.tools.assert_equal(RHUIManagerRepo.list(connection), [])
+
+def test_03_remove_existing_entitlement_certificates():
+    '''Clean up uploaded entitlement certificates'''
+    RHUIManager.remove_rh_certs(connection)
+
+def test_04_create_custom_repo():
+    '''Create a custom repo for further testing (interactively; not currently supported by the CLI)'''
+    RHUIManagerRepo.add_custom_repo(connection, custom_repo_name, entitlement="n")
+
+def test_05_upload_entitlement_certificate():
+    '''Upload the Atomic (the small) entitlement certificate'''
+    RHUIManagerCLI.cert_upload(connection, "/tmp/extra_rhui_files/rhcert_atomic.pem", "Atomic")
+
+def test_06_check_certificate_info():
+    '''Check certificate info for validity'''
+    RHUIManagerCLI.cert_info(connection)
+
+def test_07_check_certificate_expiration():
+    '''Check if the certificate expiration date is OK'''
+    RHUIManagerCLI.cert_expiration(connection)
+
+def test_08_check_unused_product():
+    '''Check if a repo is available'''
+    RHUIManagerCLI.repo_unused(connection, yum_repo_name_1)
+
+def test_09_add_rh_repo_by_product():
+    '''Add a Red Hat repo by its product name'''
+    RHUIManagerCLI.repo_add(connection, yum_repo_name_1)
+
+def test_10_add_rh_repo_by_id():
+    '''Add a Red Hat repo by its ID'''
+    RHUIManagerCLI.repo_add_by_repo(connection, [yum_repo_id_2])
+
+def test_11_repo_list():
+    '''Check the added repos'''
+    RHUIManagerCLI.repo_list(connection, yum_repo_id_1, yum_repo_name_1)
+    RHUIManagerCLI.repo_list(connection, yum_repo_id_2, yum_repo_name_2)
+
+def test_12_start_syncing_repo():
+    '''Start syncing one of the repos'''
+    RHUIManagerCLI.repo_sync(connection, yum_repo_id_2, yum_repo_name_2)
+
+def test_13_repo_info():
+    '''Verify that the repo name is part of the information about the specified repo ID'''
+    RHUIManagerCLI.repo_info(connection, yum_repo_id_2, yum_repo_name_2)
+
+def test_14_check_package_in_repo():
+    '''Check a random package in the repo'''
+    RHUIManagerCLI.packages_list(connection, yum_repo_id_2, "ostree")
+
+def test_15_upload_rpm_to_custom_repo():
+    '''Upload content to the custom repo'''
+    RHUIManagerCLI.packages_upload(connection, custom_repo_name, "/tmp/extra_rhui_files/rhui-rpm-upload-test-1-1.noarch.rpm")
```

Comment:
@RadekBiba move it after creat_custom_repo test.

Reply:
@RadekBiba missing tests: check repo id uniqueness, upload several rpms, check rpms are in the repo.

Reply:
1. OK
2. The uniqueness check is already done in test_repo_management.py, so let's not duplicate the additional tests. (I would be fine with this if the CLI was able to create custom repos, but that's not the case). Checking that the rpm is indeed in the custom repo after completing the upload is a good idea, though, and I'll add that.